watertap/property_models/tests/test_coag_prop_pack.py

```diff
@@ -0,0 +1,271 @@
+###############################################################################
+# ProteusLib Copyright (c) 2021, The Regents of the University of California,
+# through Lawrence Berkeley National Laboratory, Oak Ridge National
+# Laboratory, National Renewable Energy Laboratory, and National Energy
+# Technology Laboratory (subject to receipt of any required approvals from
+# the U.S. Dept. of Energy). All rights reserved.
+#
+# Please see the files COPYRIGHT.md and LICENSE.md for full copyright and license
+# information, respectively. These files are also available online at the URL
+# "https://github.com/nawi-hub/proteuslib/"
+#
+###############################################################################
+import pytest
+from watertap.property_models.coagulation_prop_pack import (CoagulationParameterBlock,
+                                                            CoagulationStateBlock)
+from watertap.property_models.tests.property_test_harness import PropertyAttributeError
+from pyomo.environ import (ConcreteModel,
+                           assert_optimal_termination,
+                           value,
+                           Set,
+                           Param,
+                           Var,
+                           units as pyunits,
+                           Suffix,
+                           Constraint,
+                           SolverFactory,
+                           SolverStatus,
+                           TerminationCondition)
+from idaes.core import (FlowsheetBlock,
+                        MaterialFlowBasis,
+                        MaterialBalanceType,
+                        EnergyBalanceType)
+from idaes.core.util.model_statistics import degrees_of_freedom
+from pyomo.util.check_units import assert_units_consistent
+import idaes.core.util.scaling as iscale
+from idaes.core.util.scaling import badly_scaled_var_generator
+from idaes.core.util import get_solver
+
+__author__ = "Austin Ladshaw"
+
+solver = get_solver()
+
+# -----------------------------------------------------------------------------
+# Start test class
+class TestCoagulationPropPack():
```

Comment:
Did you try using the WaterTAP property test harness? It has a set of standard tests that are helpful. For example, it would have caught that the density constraint was not transformed for scaling.

Reply:
I don't know how to use it and I don't really have time to mess with it at the moment. This needs to get in ASAP because it is holding up the unit model that has to be in by March (and I really don't have the time right now).